### PR TITLE
HCE-794: Add project_id to Boundary Cluster Resource

### PR DESCRIPTION
**Pull request by paraspraj**
### :hammer_and_wrench: Description

Add project_id to Boundary Cluster resource.

### CODE REVIEW

File: internal/provider/data_source_boundary_cluster.go [hunk 1]
```diff
@@ -25,6 +26,14 @@ func dataSourceBoundaryCluster() *schema.Resource {
 				Required:         true,
 				ValidateDiagFunc: validateSlugID,
 			},
+			// Optional inputs
+			"project_id": {
+				Description:  "The ID of the HCP project where the Boundary Cluster is located.",
```
Comment (bcmdarroch) [inline internal/provider/data_source_boundary_cluster.go:31]:
```suggestion
				Description:  "The ID of the HCP project where the Boundary cluster is located.",
```

File: internal/provider/resource_boundary_cluster.go [hunk 2]
```diff
@@ -62,6 +63,15 @@ func resourceBoundaryCluster() *schema.Resource {
 				ValidateDiagFunc: validateBoundaryPassword,
 				Sensitive:        true,
 			},
+			// Optional inputs
+			"project_id": {
+				Description:  "The ID of the HCP project where the Boundary Cluster is located.",
```
Comment (bcmdarroch) [inline internal/provider/resource_boundary_cluster.go:68]:
```suggestion
				Description:  "The ID of the HCP project where the Boundary cluster is located.",
```

**Review by bcmdarroch — APPROVED** [review]

File: internal/provider/data_source_boundary_cluster.go [hunk 3]
```diff
@@ -51,11 +60,16 @@ func dataSourceBoundaryCluster() *schema.Resource {
 func dataSourceBoundaryClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 
 	client := meta.(*clients.Client)
-
 	clusterID := d.Get("cluster_id").(string)
+
+	projectID, err := GetProjectID(d.Get("project_id").(string), client.Config.ProjectID)
```
[the remark this answers is not in the document]
Comment (paraspraj) [reply internal/provider/data_source_boundary_cluster.go:65]: Hey @guygrigsby! So we do have some logic in the SDK helper that checks for the field over here in the `terraform-plugin-sdk`: https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/resource_data.go#L84

As a sanity check I tried to Terraform Apply two Boundary resources.
```
resource "hcp_boundary_cluster" "example" {
  cluster_id = "boundaryid"
  username = "test"
  password = "testtest"
}
```
In this case, the Plan ran and created the cluster without issue.

```
resource "hcp_boundary_cluster" "example" {
  cluster_id = "boundaryid"
  username = "test"
  password = "testtest"
  project_id = ""
}
```
In this case, the Plan returned a nicely formatted invalid UUID error to the user.

[the remark this answers is not in the document]
Comment (guygrigsby) [reply internal/provider/data_source_boundary_cluster.go:65]: Hi @paraspraj thank you the explanation and for double checking! 😄

**Review by guygrigsby — APPROVED** [review]
I am not sure my approval means anything, but here it goes! 💨

**Pull request merged**